### PR TITLE
💡Siri UI for creating custom time entry

### LINKED ISSUE #4360: Create Siri shortcuts for completely custom entries
Until know Siri shortcuts are created automatically from whatever the user does in the app.

The idea with this issue is to allow the user to manually define how the entry he'll create with Siri will be: description, project, tags...

We'll need a UI for that, which will probably reside in settings together with the list of shortcuts.

I'm attaching the way Things does this as a reference:

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
As the title says.

### Relationships
Closes #4360

## How is it done?
Similar to how the custom report is done https://github.com/toggl/mobileapp/issues/4795

### Why not another way?
I'm not particularly happy about the ViewModel exposing that much information. However I coulnd't really find another cleaner way to do it. 

### Side effects
New screen so none.

## Review hints
None beside testing it out.

## :squid: Permissions
🎙

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/Settings/Siri/SiriShortcutsCustomTimeEntryViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        public BehaviorRelay<bool> IsBillable = new BehaviorRelay<bool>(false);
+        public BehaviorRelay<string> Description = new BehaviorRelay<string>(string.Empty);
+        public BehaviorRelay<IThreadSafeWorkspace> Workspace = new BehaviorRelay<IThreadSafeWorkspace>(null);
+        public BehaviorRelay<long?> ProjectId = new BehaviorRelay<long?>(null);
+        public BehaviorRelay<long?> TaskId = new BehaviorRelay<long?>(null);
+        public BehaviorRelay<IEnumerable<IThreadSafeTag>> Tags =
+            new BehaviorRelay<IEnumerable<IThreadSafeTag>>(Enumerable.Empty<IThreadSafeTag>());
+        public IObservable<bool> IsBillableAvailable { get; }
+        public IObservable<IEnumerable<string>> TagsName { get; }
```
Comment (Odrakir) [inline Toggl.Core.UI/ViewModels/Settings/Siri/SiriShortcutsCustomTimeEntryViewModel.cs]: `TagNames` maybe?

File: Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                    new Color(info.ProjectColor).ToNativeColor()))
+                .Subscribe(ProjectTaskClientLabel.Rx().AttributedText())
+                .DisposedBy(DisposeBag);
+
+            ViewModel.ProjectClientTask
+                .Select(info => info.HasProject)
+                .Subscribe(ProjectTaskClientLabel.Rx().IsVisible())
+                .DisposedBy(DisposeBag);
+
+            ViewModel.ProjectClientTask
```
Comment (Odrakir) [inline Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs]: I would probably use just one subscription (maybe with a private method) for these three subscriptions to `ViewModel.ProjectClientTask`. But I'm just being picky, you can leave it like it is. Your choice.

File: Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+            DescriptionTextView.TintColor = Colors.StartTimeEntry.Cursor.ToNativeColor();
+            DescriptionTextView.PlaceholderText = Resources.AddDescription;
+        }
+
+        private void centerTextVertically(UITextView textView)
+        {
+            var topOffset = (textView.Bounds.Height - textView.ContentSize.Height) / 2;
+            textView.ContentInset = new UIEdgeInsets(topOffset, 0, 0, 0);
+        }
```
Comment (Odrakir) [inline Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs]: new line

**Comment by khoi** [comment]
image

the icon isn't right, I'll update it once I get the asset from designer

File: Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs [hunk 4]
```diff
@@ -97,13 +99,51 @@ public override void ViewDidLoad()
...
+                {
+                    if (enabled)
+                    {
+                        DescriptionTextView.ResignFirstResponder();
+                    }
+                    else
+                    {
+                        DescriptionTextView.BecomeFirstResponder();
+                    }
+                });
```
Comment (Odrakir) [inline Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs]: DisposeBag?

File: Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs [hunk 5]
```diff
@@ -182,10 +224,52 @@ private void prepareSiriButton()
...
+                ? constructStartTimerFromClipboardIntent()
+                : constructStartTimerIntent();
+
+            var vc = new INUIAddVoiceShortcutViewController(new INShortcut(intent));
+            vc.ModalPresentationStyle = UIModalPresentationStyle.FormSheet;
+            vc.Delegate = this;
+            PresentViewController(vc, true, null);
+        }
+
+        private INIntent constructStartTimerFromClipboardIntent()
```
Comment (Odrakir) [inline Toggl.iOS/ViewControllers/Settings/Siri/SiriShortcutsCustomTimeEntryViewController.cs]: I think this should be merged with `constructStartTimerIntent` receiving a parameter.

**Pull request merged**